packages/core/src/util/tokenTransferInspector.ts

```diff
@@ -148,11 +157,37 @@ const toTokenTransferValue = async (
   return tokenTransferValue;
 };
 
+/**
+ * Resolve inputs with a timeout so that if the inputs fail to resolve in n Miliseconds
+ * we throw an error so the UI consumer can handle situation accordingly.
+ * In this case we create a race condition in which we race to send the resolved inputs before the
+ * timeout, and timeout only if the inputs aren't there.
+ */
+export const resolveWithTimeout = <T>(promise: Promise<T>, timeout: Milliseconds): Promise<T> =>
+  Promise.race([
+    promise,
+    new Promise<T>((_, reject) =>
+      setTimeout(() => reject(new Error('Timeout: failed to resolve inputs in time')), timeout)
```

Comment:
A more specifically-typed error would be nice
```suggestion
      setTimeout(() => reject(new TimeoutError('Timeout: failed to resolve inputs in time')), timeout)
```

Reply:
✅